packages/presentation/presentation_publishing/interfaces/fetch/fetch.test.ts

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { AggregateQuery, Filter, Query, TimeRange } from '@kbn/es-query';
+import { BehaviorSubject, skip, Subject } from 'rxjs';
+import { fetch$ } from './fetch';
+
+describe('onFetchContextChanged', () => {
+  const onFetchMock = jest.fn();
+  const parentApi = {
+    filters$: new BehaviorSubject<Filter[] | undefined>(undefined),
+    query$: new BehaviorSubject<Query | AggregateQuery | undefined>(undefined),
+    reload$: new Subject<void>(),
+    searchSessionId$: new BehaviorSubject<string | undefined>(undefined),
+    timeRange$: new BehaviorSubject<TimeRange | undefined>(undefined),
+    timeslice$: new BehaviorSubject<[number, number] | undefined>(undefined),
+  };
+
+  beforeEach(() => {
+    onFetchMock.mockReset();
+    parentApi.filters$.next(undefined);
+    parentApi.query$.next(undefined);
+    parentApi.searchSessionId$.next(undefined);
+    parentApi.timeRange$.next(undefined);
+    parentApi.timeslice$.next(undefined);
+  });
+
+  describe('searchSessionId', () => {
+    let i = 0;
+    function setSearchSession() {
+      i++;
+      parentApi.searchSessionId$.next(`${i}`);
+    }
+    beforeEach(() => {
+      i = 0;
+      setSearchSession();
+    });
+
+    test('should emit on subscribe when only searchSession is provided', async () => {
+      const api = {
+        parentApi: {
+          searchSessionId$: parentApi.searchSessionId$,
+        },
+      };
+      const subscription = fetch$(api).subscribe(onFetchMock);
+      await new Promise((resolve) => setTimeout(resolve, 0));
+      expect(onFetchMock.mock.calls).toHaveLength(1);
+      const fetchContext = onFetchMock.mock.calls[0][0];
+      expect(fetchContext.searchSessionId).toBe('1');
+      subscription.unsubscribe();
+    });
+
+    test('should emit once on fetch context changes', async () => {
+      const subscription = fetch$({ parentApi }).pipe(skip(1)).subscribe(onFetchMock);
+      await new Promise((resolve) => setTimeout(resolve, 0));
+      expect(onFetchMock.mock.calls).toHaveLength(0);
+
+      parentApi.filters$.next([]);
+      parentApi.query$.next({ language: 'kquery', query: '' });
+      parentApi.timeRange$.next({
+        from: 'now-24h',
+        to: 'now',
+      });
+      parentApi.timeslice$.next([0, 1]);
+      setSearchSession();
+
+      await new Promise((resolve) => setTimeout(resolve, 10));
```

Comment:
Why are we waiting for 10ms here? Shouldn't the debounce only cause 1ms of delay?

Reply:
I will change `0`. Its left over from some testing